File: javascript/lib/envbyjson.test.ts
import { describe, expect, test } from "@jest/globals";

import envbyjson from "./envbyjson";

describe("envbyjson module", () => {
  test("load", () => {
    envbyjson.load("lib/testdata/env.json");
    expect(process.env.A).toBe("got A");
    expect(process.env.B).toBe("got B");
    envbyjson.load("lib/testdata/env2.json", "lib/testdata/env3.json");
    expect(process.env.C).toBe("got C");
    expect(process.env.D).toBe("got D");
  });

  test("load wrong", () => {
    const t = () => {
      envbyjson.load("lib/testdata/env_err.json");
    };
    expect(t).toThrow(Error);
  });

  test("loadProp", () => {
    envbyjson.loadProp("lib/testdata/env4.json", "lib/testdata/env5.json", "P");
    expect(process.env.H).toBe("got H");
    expect(process.env.I).toBe("got I");
  });

  test("loadProp wrong", () => {
    const t = () => {
      envbyjson.loadProp("lib/testdata/env4_err.json", "P");
    };
    expect(t).toThrow(Error);
  });
});

File: go/envbyjson_test.go
package envbyjson

import (
	"os"
	"testing"
)

func TestDefaultLoad(t *testing.T) {
	err := Load()
	if err != nil {
		t.Error(err)
	}
	resultA := os.Getenv("A")
	resultB := os.Getenv("B")
	if resultA != "got A" || resultB != "got B" {
		t.Error("properties are not loaded correctly")
	}
}

func TestLoadWrongType(t *testing.T) {
	err := Load("testdata/env_err.json")
	if err == nil {
		t.Errorf("non-string property must cause an error")
	}
}

func TestLoadGivenPath(t *testing.T) {
	err := Load("testdata/env2.json")
	if err != nil {
		t.Error(err)
	}
	resultA := os.Getenv("A")
	resultB := os.Getenv("B")
	if resultA != "got A" || resultB != "got B" {
		t.Error("properties are not loaded correctly")
	}
}

func TestLoadProp(t *testing.T) {
	err := LoadProp("testdata/env3.json", "C")
	if err != nil {
		t.Error(err)
	}
	resultD := os.Getenv("D")
	resultE := os.Getenv("E")
	if resultD != "got D" || resultE != "got E" {
		t.Error("properties are not loaded correctly")
	}
}

func TestLoadPropNoParam(t *testing.T) {
	err := LoadProp()
	if err == nil {
		t.Error("at least one param must be given")
	}
}

func TestLoadPropWrongType(t *testing.T) {
	err := LoadProp("testdata/env3_err.json", "C")
	if err == nil {
		t.Errorf("non-string property must cause an error")
	}
}
